Makefile

```diff
@@ -103,3 +103,16 @@ binaries-all-sys:
 	GOOS=linux   GOARCH=amd64 $(MAKE) binaries
 	GOOS=linux   GOARCH=arm64 $(MAKE) binaries
 	GOOS=windows GOARCH=amd64 $(MAKE) binaries
+
+.PHONY: update-dep
+update-dep:
+	for mod in $$(grep -l '$(MODULE) ' $$(find . -name go.mod)); do \
```

Comment:
Use for-all target

Reply:
`$(MAKE) for-all CMD="go get -u $(MODULE)@$(VERSION)"`

Reply:
Don't think we want -u, at least for otel-collector, if I understand docs correctly.

Also it needs to check if the module is already present so that it doesn't add it to modules that don't use it. To do that and use for-all I had to make it an external script.

Reply:
> The -u flag instructs get to use the network to update the named packages and their dependencies. By default, get uses the network to check out missing packages but does not use it to look for updates to existing packages.

For me it looks like you need -u because otherwise it updates with the version from GOPATH.

Reply:
It wouldn't be looking in GOPATH if it's a module project I don't think. Seems like the doc you reference predates modules? `go help get` says:

```
Although get defaults to using the latest version of the module containing
a named package, it does not use the latest version of that module's
dependencies. Instead it prefers to use the specific dependency versions
requested by that module. For example, if the latest A requires module
B v1.2.3, while B v1.2.4 and v1.3.1 are also available, then 'go get A'
will use the latest A but then use B v1.2.3, as requested by A. (If there
are competing requirements for a particular module, then 'go get' resolves
those requirements by taking the maximum requested version.)

The -u flag instructs get to update modules providing dependencies
of packages named on the command line to use newer minor or patch
releases when available. Continuing the previous example, 'go get -u A'
will use the latest A with B v1.3.1 (not B v1.2.3). If B requires module C,
but C does not provide any packages needed to build packages in A
(not including tests), then C will not be updated.
```

So the question is do we want to use the same dependencies as otel-col or do we want to upgrade to latest? I would argue we should same dependencies as otel-col.